### ISSUE TITLE
nk_edit_string accepts backspace, but no other input

### ISSUE THREAD

**Issue by Atemboi** [vurtun/nuklear#446]
Hi, I was adding nk_edit_string into one of my windows, however, when I do so, I can only delete text and not append any new text to the string.

I am doing this:
`nk_edit_string_zero_terminated(ctx,NK_EDIT_SIMPLE,text,10,nk_filter_default);` to no avail

**Comment by vurtun** [comment]
It seems to work fine for me. I can add and remove text without problem. You are passing `10` as argument for maximum number of bytes to be written are you sure this is correct?

**Comment by Atemboi** [comment]
yeah, that's incorrect, I'm using ``sizeof`` now, but it still does not accept anything else

**Comment by vurtun** [comment]
Could you post a complete code sample for me. I need an overview of what you are doing so that I can reproduce the problem (does not have to be functioning just so that I get an overview).

**Comment by wheybags** [comment]
Are you using one of the example backends or your own one? If you're using your own, are you handling text input properly? Backspace and a few other keys are handled explicitly with nk_input_key in the example backends, but text input is handled separately, to allow for non-ascii input. (See https://github.com/vurtun/nuklear/blob/master/demo/sdl_opengl3/nuklear_sdl_gl3.h#L419 )

If you only copy pasted in the explicit key handling from one of the backends like this, then backspace would work, but normal text wouldn't, just like you described.

**Comment by noxytrux** [comment]
@Atemboi Make sure you poll events between nk_input_begin/nk_input_end eg:

```
void text_callback(GLFWwindow *win, unsigned int codepoint) {

    nk_input_unicode(ctx, codepoint);
}

glfwSetCharCallback(win, text_callback);

nk_input_begin(ctx);
glfwPollEvents(); 
nk_input_end(ctx);
```
